### PR TITLE
Feature

### PR DESCRIPTION
**Pull request by pacyL2K19**
# Advanced Enumerables Solution.
We built some of the ruby's own enumerable methods like:

my_each
my_each_with_index
my_select
my_all?
e. t. c...

### CODE REVIEW

**Review by MahmoudBakr23 — CHANGES_REQUESTED** [review]
### **Required Changes ♻**

<hr>

Hello @Gnwin and @pacyL2K19 👋
Great job working on this project. However, there are mandatory changes to do before submitting

- [x] Please add a well-descriptive Pull Request header
- [x] Please add a well-descriptive Pull Request body
- [x] Please make sure to set up a code linter and to run and pass rubocop checks on the pull request and follow the Git Flow
Screenshot1

- [x] Please add a multiply_els method to test my_inject method
multiply_els


<hr>

Please make sure to do all of the changes above before you request another code review.
Good luck and happy coding 💻👨‍💻

File: enumerables.rb [hunk 1]
```diff
@@ -0,0 +1,313 @@
...
+          result = result.send(acc, el)
+        end
+      end
+    end
+    result
+  end
+  
+  def multiply_els(num = 1)
+    self.my_inject(num) { |product, n| product * n }
+  end
```
Comment (LamiaSristy) [inline enumerables.rb]: Please Remove this code block from the module Enumerable and place this outside the module.
Currently, to run the code You have to run => ([2, 4, 5]).multiply_els().
Consider this writing it this way:

``` 
   def multiply_els(arr)
      arr.my_inject(1, '*')
   end

**Review by LamiaSristy — CHANGES_REQUESTED** [review]
## Required Changes 🧧 
Hello @Gnwin and @pacyL2K19 👋
Great work so far on this project. You have done all the required changes by the previous reviewer. 👍 👍 
However, there is only one thing that needs to change.
* [x] In your multiply_els method you need to place it outside the module Enumerable. As now  ```multiply_els([2, 4, 5]) #=> 40``` is not working, as described on Odin ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/multiply_els.png) from [Odin](https://github.com/TheOdinProject/curriculum/blob/master/ruby_programming/archive/basic_ruby/project_advanced_building_blocks.md#project-2-enumerable-methods)).
Please make these changes and again ask for a re-review. 🙏 🙏 
If you have any queries about this review, you can reach me at @lamiasristy
Good Luck. 👍 👍

**Comment by pacyL2K19** [comment]
Thank you for your review, I made the required change, hopefully, everything is now set

**Review by Georjane — APPROVED** [review]
## Status: PR Approved ✔️ 💯
### Hi Team 🙌
Nicely done.👏 Congratulations on successfully completing the Enumerable project. You have implemented all changes required by the previous TSE and all tests are passing. It's a thumbs up for me.👍
You can go ahead and merge this PR. 

Good luck on your next project and Happy Coding 💻

**Pull request merged**